docs/en/deploy/Running-Alluxio-On-Yarn.md

```diff
@@ -6,16 +6,20 @@ group: Deploying Alluxio
 priority: 6
 ---
 
-TODO: Expand the explanation of why we prefer to run alongside YARN instead of on top. Relevant
-mailing list thread: https://groups.google.com/forum/#!topic/alluxio-users/KEvBE9AiJlE
-TODO: Tell users how to configure Alluxio servers running on Yarn
-TODO: Discuss how to configure the YARN application classpath so that applications can work with Alluxio.
-TODO: Add troubleshooting help
-- How to debug when alluxio-yarn.sh fails to bring up Alluxio servers
-- How to find logs when Alluxio containers fail to start
-- How to diagnose when there aren't enough YARN resources to launch Alluxio containers
+This document discusses how to run Alluxio in a YARN environment. There are two
+approaches: [running Alluxio alongside YARN](#running-alluxio-alongside-yarn), and
+[running Alluxio as an application on top of YARN](#running-alluxio-as-a-yarn-application).
+We support both approaches, but recommend running Alluxio alongside YARN for a few reasons:
 
-## Standalone
+- Alluxio servers are long-running daemons like HDFS. HDFS is not supposed to run on YARN.
+- Alluxio's architecture is designed to have one worker per host, but this is not
+well supported by YARN.
+- Running Alluxio on top of YARN adds the additional complexity of managing Alluxio's
+ApplicationMaster.
+- Multi-master high availability mode is not currently supported by our YARN integration.
+- The YARN integration does not support restarting workers in case they crash.
+
+## Running Alluxio Alongside YARN
```

Comment:
To run Alluxio alongside YARN, we need to ensure YARN apps can talk to Alluxio. What's the requirement or common practice? e.g., do we need to set the "yarn.application.classpath" property to include Alluxio client jar?

Reply:
added